src/mongo/connectToMongoClient.ts

```diff
@@ -13,5 +14,19 @@ export async function connectToMongoClient(connectionString: string, extensionUs
         appname: extensionUserAgent
     };
 
-    return await MongoClient.connect(connectionString, options);
+    try {
+        return await MongoClient.connect(connectionString, options);
+    } catch (err) {
+        let error = <{ message?: string, name?: string }>err;
+        let name = error && error.name;
+        let message = error && error.message;
+
+        // Example error: "failed to connect to server [localhost:10255] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:10255]"
+        // Example error: "failed to connect to server [127.0.0.1:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]"
+        if (name === 'MongoError' && /ECONNREFUSED/.test(message) && /(localhost|127\.0\.0\.1)/.test(message)) {
```

Comment:
I can't because that brings in vscode from the ui lib which doesn't work in the context of the language server, which uses this... :-)   (Originally did it that way.)